### PR TITLE
feature(multimedia) add name validation

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description
Add multimedia name validation prevent form using '/' character

Issue #1637
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/migration/src/AbstractErgonodeMigration.php [hunk 1]
```diff
@@ -20,6 +20,6 @@ abstract class AbstractErgonodeMigration extends AbstractMigration
      */
     public function down(Schema $schema): void
     {
-        $this->throwIrreversibleMigrationException();
+        //$this->throwIrreversibleMigrationException();
```
Comment (piotrkreft) [inline module/migration/src/AbstractErgonodeMigration.php:23]: Comment for removal

File: module/product/src/Infrastructure/Grid/ProductGridBuilder.php [hunk 2]
```diff
@@ -141,7 +141,7 @@ public function build(GridConfigurationInterface $configuration, Language $langu
                 'parameters' => ['language' => $language->getCode(), 'product' => '{id}'],
                 'method' => Request::METHOD_DELETE,
             ],
-        ]))->orderBy('index', 'DESC');
+        ]))->orderBy('esa_index', 'DESC');
```
Comment (piotrkreft) [inline module/product/src/Infrastructure/Grid/ProductGridBuilder.php:144]: This is done via a different commit already. Perhaps merge would be in place if needed

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — DISMISSED** [review]

**Review by wiewiurdp — DISMISSED** [review]

File: module/multimedia/src/Application/Validator/MultimediaNameValidator.php [hunk 3]
```diff
@@ -28,13 +27,14 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        if (!$value instanceof UploadedFile) {
-            throw new UnexpectedTypeException($value, UploadedFile::class);
-        }
+        $value = (string) $value;
 
-        if ($constraint->max < strlen($value->getClientOriginalName())) {
+        if (128 < strlen($value)) {
```
Comment (wfajczyk) [inline module/multimedia/src/Application/Validator/MultimediaNameValidator.php:32]: Please use const 128

File: module/multimedia/src/Application/Validator/MultimediaUploadNameValidator.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!$value instanceof UploadedFile) {
+            throw new UnexpectedTypeException($value, UploadedFile::class);
+        }
+
+        if (128 < strlen($value->getClientOriginalName())) {
```
Comment (wfajczyk) [inline module/multimedia/src/Application/Validator/MultimediaUploadNameValidator.php:35]: Pleasue use const 128

**Review by wfajczyk — CHANGES_REQUESTED** [review]

**Review by wfajczyk — DISMISSED** [review]

**Review by wfajczyk — APPROVED** [review]
